First Release 4.0.1.0

RinaDev-xyz opened the pull request:
This is going to be the main 4.0.1.0 before release I want to do the following:

###  Main Development stuff:

- [x] Have the translation system fully working.
- [x] Translation system is able to safe user preferences.
- [x] Removed doubles system.
- [x] Better combines materials, current system is a bit wonky.
- [x] Separate Meshes by materials and loose parts.
- [x] Able to merge bones.
- [x] Able to delete zero weight bones and constraints.
- [x] Updater system so users can just update the plugin in blender.

### Finale tasks:

- [x] Organize UI.
- [x] Work on merge bones more and ensure child bones don't show up as often. 
- [x] Clean up code, actually use helper functions which are in common.py as much as possible and etc.
- [x] Test plugin.
- [x] Fix any issues.
- [x] Update wiki.

RinaDev-xyz committed:
Translation system complete, plugin settings.

- The translations system is no complete, the plugin will now remember what language you had selected.
- Buttons labels and description now get translated correctly.
- New addon preferences system, this will allow the plugin to save the language you had selected and load it next time you start blender. this will allow us to save other stuff in the future as well in a setting.json the plugin creates on start up.

RinaDev-xyz committed:
Removed Doubles Functions

- Adds the remove doubles functions, this will remove doubles, however it will avoid meshes with shapekey, safer then the blender version but will remove less doubles.
- Buttons will not be clickable unto a armature is in the scene.

RinaDev-xyz committed:
Material Combiner Updates

- Material Combiner will try and merge textures as well.
- Material Combiner will try and consolidate nodes.
- Fixed issue where removing materials  was only working when material where selected. Now it will select all meshes and do this automatically.

RinaDev-xyz committed:
Seperate By Materials and Mehses

Adds Seperate By Materials and Mehses

RinaDev-xyz committed:
Fix to translations

RinaDev-xyz committed:
Update README.md

RinaDev-xyz committed:
Update README.md

RinaDev-xyz committed:
This is start of the merge bone system.

This is the start of the merge bones system, still needs a bit of work for it to be fully functional.

The basic comecpt that it will give of list of parent bones which are not in the exclude list. you then be able to merge a ratio of the bones which will help lower bones count for Pys bones in vrchat and etc.

Again this is basic and bare bones and needs work still.

RinaDev-xyz committed:
Remove Doubles Re-Do

I have re-done the remove doubles code, it works so much better now and does a much better job at preserving shape keys.

RinaDev-xyz committed:
Merge Bones Re-make

Got rid of the old version, re-made it now.
- Currently it allows you to merge any bones however in the future I going to exclude common bones.
- UI is messed at the minute will fix later.

Yusarina commented:
Merge Bones is added and finally works, however it still needs alot of work as we don't want to allow users to merge all there bones.... will add something to stop this later.

RinaDev-xyz committed:
Fixed issue where we were removing the base bone, we 100% should not do that.

RinaDev-xyz committed:
Merge Bones Major Changes

Made quite a few changes to merge bones:

- We now exclude most important bones to prevent accidently merging then this includes hips, chest, thumb, head, neck, twist bones, Eyes, Tongue, Fingers, Shoulders, Arms, Elbow, Leg, Knee, Toe, Teeth and Hand bones.
- We ensure both upper case and lower case bones are excluded.
- I also started to remove bones with 2, .002, _002 + as well, though this system is not perfect and will need more work (Or a rethink).
- We no longer use the search system and just list the bones you can merge.
- Added error reporting so you know what went wrong in certain circumstances.
- Better merge logic.

Though this is not perfect, it's better then it was, however I still want to work on it before the first main release as I want more bones to be removed from the list so models with large amount of bones won't have 100s listed.

Yusarina commented:
Ok merge bones is in a much better state then it was so i marked it off, however it still needs some work so I added Work on merge bones more and ensure child bones don't show up as often to my finale tasks!

RinaDev-xyz committed:
First Remove Zero Bones Function.

RinaDev-xyz committed:
Remove Zero Wight Bones Updated

Works correctly now. Large changes and is more reliable and will report when works/ not.

RinaDev-xyz committed:
Remove Bone constraints

RinaDev-xyz committed:
Start of updater no where near finish.

RinaDev-xyz committed:
Use help function in bones.py

RinaDev-xyz committed:
Join meshes use new helper functions get_meshes and unselect_all()

Join meshes uses the helper functions much more now!

RinaDev-xyz committed:
Separate Meshes Update

Separate Meshes will now unselect all before perfromacing the action. it will also fix uv cords once the operation is done.

Also updated to use helper functions and added self report to let the user know it's been done (And translated them).

RinaDev-xyz committed:
Removed text from the UI side, not needed as classes have this and was a temp work around for a bug which was fixed a bit ago.

RinaDev-xyz committed:
Translations and UI Update

- Added Translations for Bone merging and etc.
- Fixed Bone merging and other bone options not displaying in the UI correctly.

Yusarina commented:
Ok alot of the main things are complete, I just need to do the following.

1: Complete the Updater, I may use a third part thing for this instead of making a custom one as it may take too much time for me to do this.
2: Make finale changes to merge bones system, need it to be more rebust at removing child bones.
3: Organise the UI, just move a couple of things around, add text and etc. 
4: Fully Test the plugin and fix any issues found. 
5: Update the Wiki.

Once all of this is done I can release the first initial version! I hope to get all this done by Friday.

RinaDev-xyz committed:
Updater, Update

I decided that I really don't want to spend ages on a custom updater therefor using an example one from https://github.com/CGCookie/blender-addon-updater/tree/master

At the minute it will always say there is an update, that's because It's coming from a test repo, don't update it if it propts you, I just too tired to continue configing the updater tonight and will continue tomorrow.

RinaDev-xyz committed:
Fix issue when disableding the plugin

Yea, let's registry something when unregistering it?????

RinaDev-xyz committed:
Updater Update

- Check for update when opening blender.
- UI panel in settings now works.
- Basic translations for UI panel done.
- Fixed some small bugs.
- Change to official github.

Yusarina commented:
Updater is complete, we just got the finale tasks now and the first release will be out!

RinaDev-xyz committed:
Ui Fixes and changes

Yusarina commented:
UI is now fully done the only things left is:

1: Make finale changes to merge bones system, need it to be more rebust at removing child bones.
2: Fully Test the plugin and fix any issues found.
3: Update the Wiki.

RinaDev-xyz committed:
Merge Bone List Fixes

RinaDev-xyz committed:
Small Fixes

Yusarina commented:
The plugin is about ready to be released, all I have to do now is update the Wiki! I get this done tomorrow and release the first version!

RinaDev-xyz committed:
Update README.md
